pkg/scheduler/predicates/ha.go

```diff
@@ -144,53 +141,54 @@ func (h *ha) Filter(instanceName string, pod *apiv1.Pod, nodes []apiv1.Node) ([]
 
 	min := -1
 	minNodeNames := make([]string, 0)
-	for nodeName, podNames := range nodeMap {
-		podsCount := len(podNames)
-		maxPodsPerNode := 0
-
-		if component == label.PDLabelVal {
+	maxPodsPerNode := 0
+
+	if component == label.PDLabelVal {
```

Comment:
this `if ... else ...` block calculate the `maxPodsPerNode` var, it is only related to `replicas` var. so i move these codes out of the `for ... range nodeMap` block.

and also we want to use the `maxPodsPerNode` in error message.

@cofyc